### PR TITLE
More options for setup notifications

### PR DESCRIPTION
**Pull request by kuzmany**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | n
| New feature? | y
| Related user documentation PR URL | https://github.com/mautic/documentation/pull/134 
| Related developer documentation PR URL | n
| Issues addressed (#s or URLs) | n
| BC breaks? | n
| Deprecations? | n

#### Description:
PR added 2 new options to configuration pane:
1, 
OneSignal use gcm_sender_id as verification for web push service. 
They recommend use default value (482941778795), but in past we had to generate own gcm_sender_id  for every account.
It's better use it in configuration due backwards compatibility.
More in documentation, part. 2.2 Edit manifest.json:
https://documentation.onesignal.com/docs/web-push-sdk-setup-https
At the moment use Mautic gcm_sender_id  in code.

2, Options for disable welcome notification

notifications

#### Steps to test this PR:
1. Open http://your-mautic-domain.tld/manifest.json before apply PR
2. Apply PR, go to Configuration > Web Notification settings and enable  it
3. Change default Shared key for push notifications from documentation to your anyhting else
4. See change in url http://your-mautic-domain.tld/manifest.json
5. Disable Welcome Notification
6. Open whatever landing page and see source code, you should see something like this
notification_example_code

### CODE REVIEW

File: app/bundles/NotificationBundle/Controller/JsController.php [hunk 1]
```diff
@@ -24,9 +24,10 @@ class JsController extends CommonController
      */
     public function manifestAction()
     {
+        $gcm_sender_id = $this->factory->getParameter('gcm_sender_id', '446150739532');
```
Comment (alanhartless) [inline app/bundles/NotificationBundle/Controller/JsController.php]: Per Symfony coding standards, variables should be camel case (`$gcmSenderId`). Also, `$this->factory` is deprecated and thus we're not using it in new code. Please change that to `$this->get('mautic.helper.core_parameters')->getParameter('gcm_sender_id', '446150739532');`

File: app/bundles/NotificationBundle/EventListener/PageSubscriber.php [hunk 2]
```diff
@@ -66,6 +66,7 @@ public function onPageDisplay(PageDisplayEvent $event)
 
         $appId       = $this->coreParametersHelper->getParameter('notification_app_id');
         $safariWebId = $this->coreParametersHelper->getParameter('notification_safari_web_id');
+        $welcomenotification_enabled       = $this->coreParametersHelper->getParameter('welcomenotification_enabled');
```
Comment (alanhartless) [inline app/bundles/NotificationBundle/EventListener/PageSubscriber.php]: Same thing here for Symfony coding standards.

File: app/bundles/NotificationBundle/Form/Type/ConfigType.php [hunk 3]
```diff
@@ -78,6 +78,32 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
+                ],
+            ]
+        );
+
+        $builder->add(
+            'welcomenotification_enabled',
+            'yesno_button_group',
+            [
+                'label' => 'mautic.notification.config.form.notification.welcome.enabled',
+                'data'  => (bool) $options['data']['welcomeNotification_enabled'],
```
Comment (escopecz) [inline app/bundles/NotificationBundle/Form/Type/ConfigType.php]: I applied this PR, cleared cache, went to global configuration and got this error on this line: `Notice: Undefined index: welcomeNotification_enabled`

File: app/bundles/NotificationBundle/Config/config.php [hunk 4]
```diff
@@ -161,5 +161,7 @@
         'notification_app_id'        => null,
         'notification_rest_api_key'  => null,
         'notification_safari_web_id' => null,
+        'gcm_sender_id' => '482941778795',
```
Comment (escopecz) [inline app/bundles/NotificationBundle/Config/config.php]: Just a minor styling, but could you move the value to the row as all others?

**Comment by escopecz** [comment]
And please, provide a link to user docs PR which explains the new configuration options.

**Comment by kuzmany** [comment]
@escopecz all fixed

**Comment by kuzmany** [comment]
@escopecz documentation added

File: app/bundles/NotificationBundle/Form/Type/ConfigType.php [hunk 5]
```diff
@@ -78,6 +78,32 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
+                ],
+            ]
+        );
+
+        $builder->add(
+            'welcomenotification_enabled',
+            'yesno_button_group',
+            [
+                'label' => 'mautic.notification.config.form.notification.welcome.enabled',
+                'data'  => (bool) $options['data']['welcomeNotificationEnabled'],
```
Comment (escopecz) [inline app/bundles/NotificationBundle/Form/Type/ConfigType.php]: I'm still getting `Notice: Undefined index: welcomeNotificationEnabled` on this line. Shouldn't it be `welcomenotification_enabled` from `app/bundles/NotificationBundle/Config/config.php`? Try to remove it from your local.php to test the problem.

Comment (kuzmany) [inline app/bundles/NotificationBundle/Form/Type/ConfigType.php]: I reverted it back. I tested with index_dev.php and didn't see  any error/notice.

**Comment by escopecz** [comment]
Works for me now. Thanks! :+1:

**Pull request merged**